app/Models/Emulator.php

```diff
@@ -106,4 +128,15 @@ public function scopeActive(Builder $query): Builder
     {
         return $query->where('active', true);
     }
+
+    /**
+     * @param Builder<Emulator> $query
+     * @return Builder<Emulator>
+     */
+    public function scopeForConsole(Builder $query, int $consoleId): Builder
```

Comment:
nit: `scopeForSystem()` and `$systemId` may be more consistent with the `systems` relation name.

Reply:
yes.